pkg/controller/acmechallenges/sync.go

```diff
@@ -256,14 +267,20 @@ func (c *controller) handleFinalizer(ctx context.Context, ch *cmacme.Challenge)
 	if len(ch.Finalizers) == 0 {
 		return nil
 	}
-	if ch.Finalizers[0] != cmacme.ACMEFinalizer {
+	if otherFinalizerPresent(ch) {
 		log.V(logf.DebugLevel).Info("waiting to run challenge finalization...")
 		return nil
 	}
 
 	defer func() {
 		// call Update to remove the metadata.finalizers entry
-		ch.Finalizers = ch.Finalizers[1:]
+		for i, e := range ch.Finalizers {
```

Comment:
Maybe we can use https://pkg.go.dev/slices@go1.23.1#DeleteFunc instead here? It makes the code easier to read.